net/utils/net_chksum.c

```diff
@@ -123,4 +123,66 @@ uint16_t net_chksum(FAR uint16_t *data, uint16_t len)
 }
 #endif /* CONFIG_NET_ARCH_CHKSUM */
 
+/****************************************************************************
+ * Name: net_chksum_adjust
+ *
+ * Description:
+ *   Adjusts the checksum of a packet without having to completely
+ *   recalculate it, exactly the same as RFC 3022, Section 4.2, Page 9.
+ *
+ * Input Parameters:
+ *   chksum - points to the chksum in the packet
+ *   optr   - points to the old data in the packet
+ *   olen   - length of old data
+ *   nptr   - points to the new data in the packet
+ *   nlen   - length of new data
+ *
+ * Limitations:
+ *   The algorithm is applicable only for even offsets and even lengths.
+ ****************************************************************************/
+
+void net_chksum_adjust(FAR uint8_t *chksum,
+                       FAR const uint8_t *optr, size_t olen,
+                       FAR const uint8_t *nptr, size_t nlen)
+{
+  /* TODO: Verify this works for little- and big-endian. */
```

Comment:
Let's add `#error` in case trying to compile if for the big endian

Reply:
`NTOHS` is used, so make it as a warning.